ear-production-suite-plugins/plugins/scene/src/scene_plugin_processor.cpp

```diff
@@ -263,16 +263,20 @@ void SceneAudioProcessor::incomingMessage(std::shared_ptr<NngMsg> msg) {
 
   } else if (cmd == commandSocket->Command::SetAdmAndMappings) {
     uint32_t mapSz = (64 * 4);
-    uint32_t admSz = msg->getSize() - mapSz - 1;
-    std::vector<uint32_t> mappings(64, 0x00000000);
+    uint32_t admSz = msg->getSize() - mapSz - mapSz - 1;
+    std::vector<uint32_t> trackUidMappings(64, 0x00000000);
+    std::vector<uint32_t> objectMappings(64, 0x00000000);
     std::string admStr(admSz, 0);
     char* bufPtr = (char*)msg->getBufferPointer();
     int bufOffset = 1;  // Skip first "command" byte
-    memcpy(mappings.data(), bufPtr + bufOffset, mapSz);
+    memcpy(trackUidMappings.data(), bufPtr + bufOffset, mapSz);
+    bufOffset += mapSz;
+    memcpy(objectMappings.data(), bufPtr + bufOffset, mapSz);
     bufOffset += mapSz;
     memcpy(admStr.data(), bufPtr + bufOffset, admSz);
-    auto future = std::async(std::launch::async, [this, admStr, mappings]() {
-      recvAdmMetadata(admStr, mappings);
+
+    auto future = std::async(std::launch::async, [this, admStr, trackUidMappings, objectMappings]() {
+      recvAdmMetadata(admStr, objectMappings);
```

Comment:
Yeah that's a fair point. It's fairly simple in what it does, just isn't very readable.
In order, the message consists of;
- a "command" byte (i.e, what the data is for) = always 1 byte
- a vector of uint32_t which maps channels (vector index) to trackuids = 64 * 32bits (`mapSz`)
- another vector of uint32_t which maps channels (vector index) to audioobjectids = 64 * 32bits (`mapSz`)
- the ADM document itself (variable size)

Because the command byte and vectors are fixed size, we can figure out the size of the ADM document (`admSz`) by subtracting the fixed sizes we know from the message size. So it just skips the first byte, copies the next `mapSz` bytes in to one vector, then the next `mapSz` bytes in to the other vector, then the remaining `admSz` bytes in to the ADM document string.
Happy to tidy this if you have a suggestion?